tests/unit/services/MachineService.test.js

```diff
@@ -300,4 +309,74 @@ describe('Machine Service', () => {
         });
     });
   });
+
+  describe('Update Machine pool size', () => {
+
+    before('Set machinePoolSize at 0', function(done) {
+      return ConfigService.set('machinePoolSize', 0)
+        .then(() => {
+          return done();
+        });
+    });
+
+    after('Set machinePoolSize at 1', function(done) {
+      return ConfigService.set('machinePoolSize', 1)
+        .then(() => {
+          return MachineService.updateMachinesPool();
+        })
+        .then(() => {
+          return done();
+        });
+    });
+
+    it('Should add machines when machinePoolSize grow up', (done) => {
+      return ConfigService.set('machinePoolSize', 2)
+        .then(() => {
+          return MachineService.updateMachinesPool();
```

Comment:
This performed automatically in `ConfigController` not in `ConfigService`. If we check this in `ConfigService` driver is not initialized for the first set and generate an error.